extension/chrome/settings/modules/keyserver.ts

```diff
@@ -128,18 +128,9 @@ View.run(class KeyserverView extends View {
     const results = await this.pubLookup.attester.lookupEmails(sendAs ? Object.keys(sendAs) : [this.acctEmail]);
     for (const email of Object.keys(results)) {
       const pubkeySearchResult = results[email];
-      if (!pubkeySearchResult.pubkey) {
-        diagnosis.hasPubkeyMissing = true;
-        diagnosis.results[email] = { pubkey: undefined, match: false };
-      } else {
-        const pub = await KeyUtil.parse(pubkeySearchResult.pubkey);
-        let match = true;
-        if (!storedKeysIds.includes(pub.id)) {
-          diagnosis.hasPubkeyMismatch = true;
-          match = false;
-        }
-        diagnosis.results[email] = { pubkey: pubkeySearchResult.pubkey, match };
-      }
+      const hasMatchingKey = pubkeySearchResult.pubkeys.some(async (pubkey) => storedKeysIds.includes((await KeyUtil.parse(pubkey)).id));
```

Comment:
I'm afraid you can't put an async function into `some` -- it will simply return `Promise<boolean>` and always be considered as truthy value -- `some` won't wait for fulfillment. So you should do it either in a loop (parsing a key is a fast operation anyway), or do this fancy `race` thing https://stackoverflow.com/a/56553349 -- I think we can add it to our `extension/js/common/core/common.ts` utils.
So this case is not covered by the unit test? Can you add a test for it? As I understand, this page is invoked with `@action-open-attester-page` click?

Reply:
Nice catch. Thanks
You are right. `some` doesn't support `async`. Let me implement it.